contracts/compliance/modular/modules/TransferFeesModule.sol

```diff
@@ -0,0 +1,210 @@
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.17;
+
+import "../IModularCompliance.sol";
+import "../../../token/IToken.sol";
+import "../../../roles/AgentRole.sol";
+import "./AbstractModule.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract TransferFeesModule is AbstractModule, Ownable {
+    /// Struct of fees
+    struct Fee {
+        uint256 rate; // min = 0, max = 10000, 0.01% = 1, 1% = 100, 100% = 10000
+        address collector;
+    }
+
+    /// Mapping for compliance fees
+    mapping(address => Fee) private _fees;
+
+    /**
+    *  this event is emitted whenever a fee definition is updated for the given compliance address
+    *  the event is emitted by 'setFee'.
+    *  compliance is the compliance contract address
+    *  _rate is the rate of the fee (0.01% = 1, 1% = 100, 100% = 10000)
+    *  _collector is the collector wallet address
+    */
+    event FeeUpdated(address indexed compliance, uint256 _rate, address _collector);
+
+    error FeeRateIsOutOfRange(address compliance, uint256 rate);
+
+    error InvalidFeeCollectorAddress(address compliance, address collector);
+
+    error CollectorAddressIsNotVerified(address compliance, address collector);
+
+    /**
+    *  @dev Sets the fee rate and collector of the given compliance
+    *  @param _compliance is the compliance contract address
+    *  @param _rate is the rate of the fee (0.01% = 1, 1% = 100, 100% = 10000)
+    *  @param _collector is the collector wallet address
+    *  Only the owner of the token smart contract can call this function
+    *  Collector wallet address must be verified
+    */
+    function setFee(address _compliance, uint256 _rate, address _collector) external {
+        address tokenAddress = IModularCompliance(_compliance).getTokenBound();
+        require(msg.sender == AgentRole(tokenAddress).owner(), "only token owner can call");
+
+        if (_rate > 10000) {
+            revert FeeRateIsOutOfRange(_compliance, _rate);
+        }
+
+        if (_collector == address(0)) {
+            revert InvalidFeeCollectorAddress(_compliance, _collector);
+        }
+
+        IIdentityRegistry identityRegistry = IToken(tokenAddress).identityRegistry();
+        if (!identityRegistry.isVerified(_collector)) {
+            revert CollectorAddressIsNotVerified(_compliance, _collector);
+        }
+
+        _fees[_compliance].rate = _rate;
+        _fees[_compliance].collector = _collector;
+        emit FeeUpdated(_compliance, _rate, _collector);
+    }
+
+    /**
+    *  @dev See {IModule-moduleTransferAction}.
+    */
+    function moduleTransferAction(address _from, address _to, uint256 _value) external override onlyComplianceCall {
+        if (_from == address(0) || _to == address(0)) {
+            return;
```

Comment:
i believe this part is unreachable as the token contract will never call the `transferred` function with zero address as `_from` or `_to`, we can remove this check imo